code/modules/fishing/fish/fish_types.dm

```diff
@@ -270,17 +270,97 @@
+	var/anger = 0
+	///The lobstrosity type this matures into
+	var/lob_type = /mob/living/basic/mining/lobstrosity/juvenile/lava
+	///at which rate the crab gains maturation
+	var/growth_rate = 100 / (12 MINUTES) * 10
+
+///A chasm crab growth speed is determined by its initial weight and size, ergo bigger crabs for faster lobstrosities
+/obj/item/fish/chasm_crab/update_size_and_weight(new_size = average_size, new_weight = average_weight)
+	. = ..()
+	var/multiplier = 1
+	switch(size)
+		if(0 to FISH_SIZE_TINY_MAX)
+			multiplier -= 0.2
+		if(FISH_SIZE_SMALL_MAX to FISH_SIZE_NORMAL_MAX)
+			multiplier += 0.2
+		if(FISH_SIZE_NORMAL_MAX to FISH_SIZE_BULKY_MAX)
+			multiplier += 0.5
+		if(FISH_SIZE_BULKY_MAX to INFINITY)
+			multiplier += 0.8
+
+	if(weight < 1000)
+		multiplier -= 0.1 * round((1000 - weight) / 200)
+	else
+		multiplier += min(0.1 * round((weight - 1000) / 500), 2)
+
+	growth_rate = /obj/item/fish/chasm_crab::growth_rate * multiplier
+
+/obj/item/fish/chasm_crab/process(seconds_per_tick)
+	. = ..()
+	var/hunger = CLAMP01((world.time - last_feeding) / feeding_frequency)
+	if(health <= initial(health) * 0.6 || hunger >= 0.6) //if too hurt or hungry, don't grow.
+		anger += growth_rate * 2 * seconds_per_tick
+		return
+
+	if(!isaquarium(loc)) //can't grow outside an aquarium.
+		return
+
+	var/obj/structure/aquarium/aquarium = loc
+	if(!aquarium.allow_breeding) //the aquarium has breeding disabled
+		return
+	if(hunger >= 0.4) //I'm hungry and angry
+		anger += growth_rate * 0.6 * seconds_per_tick
+	if(!locate(/obj/item/aquarium_prop) in aquarium) //the aquarium deco is quite barren
+		anger += growth_rate * 0.25 * seconds_per_tick
+	var/fish_count = length(aquarium.get_fishes())
+	if(!ISINRANGE(fish_count, 3, AQUARIUM_MAX_BREEDING_POPULATION * 0.5)) //too lonely or overcrowded
+		anger += growth_rate * 0.3 * seconds_per_tick
+	if(fish_count <= AQUARIUM_MAX_BREEDING_POPULATION * 0.5) //check if there's enough room to maturate.
+		maturation += growth_rate * seconds_per_tick
+
+	if(maturation >= 100)
+		finish_growing()
+
+///spawn a juvenile lobstrosity on the aquarium turf
+/obj/item/fish/chasm_crab/proc/finish_growing()
+	var/mob/living/basic/mining/lobstrosity/juvenile/lob = new lob_type(get_turf(src))
+	for(var/trait_type in fish_traits)
+		var/datum/fish_trait/trait = GLOB.fish_traits[trait_type]
+		trait.apply_to_mob(lob)
+	if(!prob(anger))
+		lob.AddElement(/datum/element/ai_retaliate)
+		lob.ai_controller.planning_subtrees -= SSai_controllers.ai_subtrees[/datum/ai_planning_subtree/simple_find_target]
+		lob.ai_controller.planning_subtrees += SSai_controllers.ai_subtrees[/datum/ai_planning_subtree/target_retaliate]
+	else if(anger < 30) //not really that mad, just a bit unstable.
+		lob.ai_controller.planning_subtrees -= SSai_controllers.ai_subtrees[/datum/ai_planning_subtree/simple_find_target]
+		lob.ai_controller.planning_subtrees += SSai_controllers.ai_subtrees[/datum/ai_planning_subtree/capricious_retaliate]
```

Comment:
These should instead be a new controller subtype. This is so that if new behaviour is introduced that directly conflicts with the personality you're trying to achieve here, it doesn't get potentially missed. It's better for ai controllers to be defined in one place rather than fractured in this way so that we don't get weird inconsistencies with behaviour.